viz/visualize.py

```diff
@@ -49,122 +54,68 @@ def __init__(self, model, dataset, model_dir=None, save_images=True,
-            # means = latent_dist[1]
-
-            heat_map_height = heat_map_size[0]
-            heat_map_width = heat_map_size[1]
-            num_latent_dims = sample_heat_map.shape[1]
-
-            heat_map = torch.zeros([num_latent_dims, 1, heat_map_height, heat_map_width])
-
-            for latent_dim in range(num_latent_dims):
-                for y_posn in range(heat_map_width):
-                    for x_posn in range(heat_map_height):
-                        heat_map[latent_dim, 0, x_posn, y_posn] = sample_heat_map[heat_map_width * y_posn + x_posn, latent_dim]
-
-            if latent_order is not None:
-                # Reorder latent samples by average KL
-                heat_map = [
-                    latent_sample for _, latent_sample in sorted(zip(latent_order, heat_map), reverse=True)
-                ]
-            heat_map_np = np.array(heat_map)
-            upsampled_heat_map = torch.tensor(upsample(input_data=heat_map_np, scale_factor=2))
-
-        # Function: all_latent_traversals
-        # Plot reconstructions in test mode, i.e. without sampling from latent
-        self.model.eval()
-        # Pass data through VAE to obtain reconstruction
-        with torch.no_grad():
-            input_data = latent_sweep_data.to(self.device)
-            sample = self.model.sample_latent(input_data)
-        # means = latent_dist[1]
-
-        avg_kl_list = read_avg_kl_from_file(os.path.join(self.model_dir, TRAIN_FILE), self.model.latent_dim)
-
-        # Perform line traversal of every latent
-        latent_samples = []
-        for idx in range(self.model.latent_dim):
-            latent_samples.append(self.latent_traverser.traverse_line(idx=idx,
-                                                                      size=size,
-                                                                      sample_latent_space=None))
-        latent_samples = [
-            latent_sample for _, latent_sample in sorted(zip(avg_kl_list, latent_samples), reverse=True)
-        ]
-
-        sorted_avg_kl_list = [
-            round(float(avg_kl_sample), 3) for avg_kl_sample, _ in sorted(zip(avg_kl_list, latent_samples), reverse=True)
-        ]
-
-        # Decode samples
-        generated = self._decode_latents(torch.cat(latent_samples, dim=0))
-
-        list_heat_map = []
-        for i in range(0, self.model.latent_dim):
-            list_heat_map.append(upsampled_heat_map[i, 0, :, :])
-
-        heat_map_sorted = [
-            list_heat_map for _, list_heat_map in sorted(zip(avg_kl_list, list_heat_map), reverse=True)
-        ]
-
-        new_torch = torch.tensor(np.zeros((self.model.latent_dim, 1, 64, 64)))
-        for i in range(0, self.model.latent_dim):
-            new_torch[i, 0, :, :] = heat_map_sorted[i]
-        nr_imgs_per_latent = size
+        image_file_name_list = [
+            os.path.join(base_directory, 'recon_comp.png'),
+            os.path.join(base_directory, 'heatmap.png'),
+            ''
```

Comment:
What is the purpose of this empty string? Why is it required?

Reply:
adapted